library/Rules/AbstractRule.php

```diff
@@ -18,7 +19,7 @@ public function __construct()
 
     public function __invoke($input)
     {
-        return !is_a($this, __NAMESPACE__.'\\NotEmpty')
+        return !$this instanceof RequiredValidatable
             && $input === '' || $this->validate($input);
```

Comment:
In the future I would like to use the `NotOptional()` instead of this hard coded checking (`$input === ''`).